cmd/openshift-install/gather.go

```diff
@@ -215,20 +247,80 @@ func logClusterOperatorConditions(ctx context.Context, config *rest.Config) erro
 		return errors.Wrap(err, "listing ClusterOperator objects")
 	}
 
+	m := make(map[string]*OperatorConditions)
 	for _, operator := range operators.Items {
 		for _, condition := range operator.Status.Conditions {
-			if condition.Type == configv1.OperatorUpgradeable {
-				continue
-			} else if condition.Type == configv1.OperatorAvailable && condition.Status == configv1.ConditionTrue {
-				continue
-			} else if (condition.Type == configv1.OperatorDegraded || condition.Type == configv1.OperatorProgressing) && condition.Status == configv1.ConditionFalse {
+			if condition.Reason == "" {
+				condition.Reason = "noreason"
+			}
+			if condition.Message == "" {
+				condition.Message = "nomessage"
+			}
+
+			switch condition.Type {
+			case configv1.OperatorAvailable:
+				if condition.Status == configv1.ConditionTrue {
+					continue
+				}
+			case configv1.OperatorDegraded:
+				if condition.Status == configv1.ConditionFalse {
+					continue
+				}
+				oc := OperatorCondition{conditionDegraded, operator.ObjectMeta.Name,
+					condition.Type, condition.Status, condition.Reason, condition.Message}
+				if m[operator.ObjectMeta.Name] == nil {
+					heap.Init(m[operator.ObjectMeta.Name])
+				}
+				heap.Push(m[operator.ObjectMeta.Name], oc)
+			case configv1.OperatorProgressing:
+				if condition.Status == configv1.ConditionFalse {
+					continue
+				}
+				oc := OperatorCondition{conditionProgressing, operator.ObjectMeta.Name,
+					condition.Type, condition.Status, condition.Reason, condition.Message}
+				if m[operator.ObjectMeta.Name] == nil {
+					heap.Init(m[operator.ObjectMeta.Name])
+				}
+				heap.Push(m[operator.ObjectMeta.Name], oc)
+			default:
```

Comment:
Exclude unknown conditions except as debugging output.  We do not need more than degraded or available, everything else is visible via a deeper dump.